### DISCUSSION TITLE
Can generateJava produce a projection that contains a connection?

### DISCUSSION THREAD

**Discussion by ghost** [Netflix/dgs-framework#758] [Q&A]
given the following schema, will the generateJava command be able to autogenerate a projection java object? Currently unable to figure this out.

```directive @connection on OBJECT

type Query {
    getPermissions(first: Int = 10, offset: Int = 0): PermissionConnection
}

type Permission @connection {
    id: ID!
    object: String!
}
```

**Comment by srinivasankavitha** [comment]
Currently, no, since the @connection directive is applied at runtime and not generated at compile time.

**Reply by ghost** [reply]
TY!
